tensorflow/python/data/util/options.py

```diff
@@ -60,6 +60,9 @@ def __setattr__(self, name, value):
                        "`tf.data.Dataset.with_options(options)` to set or "
                        "update dataset options.")
     if hasattr(self, name):
+      if name=="experimental_threading":
+        logging.warning("options.experimental_threading is deprecated."
+                        "Use options.threading instead.")
```

Comment:
An alternative approach would be to:
```
if name=="experimental_threading":
    logging.warning("options.experimental_threading is deprecated."
                    "Use options.threading instead.")
    object.__setattr__(self, "threading", value)
```

and in `dataset_ops.Options` serialize only `threading` to proto. (Although both `threading` and `experimental_threading` will be set and will always be equal during serdes).

@jsimsa WDYT?

Reply:
I like the alternative. That way we do not need to try to keep `threading` and `experimental_threading` in sync and the `experimental_threading` attribute becomes unused.